web_pivot_zoom/static/src/js/web_pivot_zoom.js

```diff
@@ -0,0 +1,205 @@
+
+openerp.web_pivot_zoom = function(instance)
+{
+    instance.web_graph.Graph.include({
+        start: function()
+        {
+            this.$el.on(
+                'click', 'td.web_pivot_zoom',
+                this.proxy('_web_pivot_zoom_on_click_cell')
+            );
+            return this._super.apply(this, arguments);
+        },
+        draw_row: function (row, frozen_rows)
+        {
+            var self = this,
+                $result = this._super.apply(this, arguments);
+            _.each(this.pivot_options.measures, function(measure, i)
+            {
+                if(!_.isEmpty(self.graph_view._web_pivot_zoom[measure.field]))
+                {
+                    jQuery($result.children().get(i + 1))
+                    .addClass('web_pivot_zoom');
+                }
+            });
+            return $result;
+        },
+        draw_measure_row: function(measure_row)
+        {
+            this._super.apply(this, arguments);
+            var self = this,
+                $last_headers = this.$thead.children().last().children();
+            _.each(this.pivot_options.measures, function(measure, i)
+            {
+                if(!_.isEmpty(self.graph_view._web_pivot_zoom[measure.field]))
+                {
+                    jQuery($last_headers.get(i + 1))
+                    .addClass('web_pivot_zoom');
+                }
+            });
+        },
+        get_measures: function()
+        {
+            var self = this;
+            return _.filter(
+                this._super.apply(this, arguments),
+                function(field)
+                {
+                    return self.pivot_options.invisible_measures.indexOf(
+                        field.field
+                    ) == -1;
+                }
+            );
+        },
+        _web_pivot_zoom_on_click_cell: function(e)
+        {
+            if(!e.currentTarget.cellIndex)
+            {
+                return;
+            }
+            var self = this,
+                $current_row = jQuery(e.currentTarget).closest('tr'),
+                $col_row = jQuery(e.currentTarget).closest('table')
+                .find('thead tr:nth-last-child(2)'),
+                col_id = $col_row.find('[data-id]').eq(
+                    Math.floor(
+                        (e.currentTarget.cellIndex - 1) /
+                        this.pivot.measures.length
+                    )
+                ).attr('data-id'),
+                row_id = $current_row.find('[data-id]').attr('data-id'),
+                current_field = this.pivot.measures[
+                    (e.currentTarget.cellIndex - 1) %
+                    this.pivot.measures.length
+                ],
+                row_header = this.pivot.get_header(row_id),
+                col_header = this.pivot.get_header(col_id),
+                measure_values = this.pivot.get_values(
+                    Math.min(col_id, row_id), Math.max(col_id, row_id)
```

Comment:
this has nothing to do with database ids, those is how the internal datastructure works: https://github.com/OCA/OCB/blob/8.0/addons/web_graph/static/src/js/pivot_table.js#L88 - I can add a link to that, but I'd expect interested readers to read this piece of code themselves

Reply:
As there are also no OCA guidelines on documenting code, it's your choice how you do it and my responsibility to study the code and supply you a PR with docstrings on every part. 

That said, other contributors and reviewers will be productive on your code much faster if the meaning of a piece of code is self-evident from the naming of the variables and functions, and if that is not possible, carry a small note of explanation. The same way you expect junior OCA contributors to polish their contribution to make things as easy as possible for the senior reviewer, the senior contributor could also strive to make life easier for junior contributors and reviewers - all in the interest of producing better code with less bugs.

Reply:
you're right. the thing is that I seriously started a commit addressing that, but what I came up with won't help, because I either reproduced the whole code in natural language, or write too little, and then you need to code anyways.
Now it's just the link above.